src/artifact/random-art-generator.cpp

```diff
@@ -383,7 +383,7 @@ static int decide_random_art_power_level(ItemEntity *o_ptr, const bool a_cursed,
     return 3;
 }
 
-static void name_unnatural_random_artifact(PlayerType *player_ptr, ItemEntity *o_ptr, const bool a_scroll, const int power_level, GAME_TEXT *new_name)
+static void name_unnatural_random_artifact(PlayerType *player_ptr, ItemEntity *o_ptr, const bool a_scroll, const int power_level, GAME_TEXT *new_name, size_t new_name_sz)
```

Comment:
The latest forced push changed that to return a std::string.  There's still a character array buffer used because of the way name_unnatural_random_artifact() gets the name.